### ISSUE TITLE
Does not support objects in an array

### ISSUE THREAD

**Issue by calebmer** [ethanresnick/json-api#43]
It would appear that json-api does not support arrays with objects inside, take the following schema:

```javascript
var discussionSchema = new Mongoose.Schema({
  messageCount: Number,
  lastIndex: Number,
  message: [{
    index: { type: Number, required: true },
    parentMessage: Number,
    author: { type: ObjectId, ref: 'User', required: true },
    text: { type: String, required: true, validate: Validators.isLength(2, 1024) },
    dateCreated: { type: Date, default: Date.now }
  }]
});
```

It errors with the following stack:

```
TypeError: Cannot read property 'name' of undefined
    at getFieldType (/node_modules/json-api/build/src/db-adapters/Mongoose/MongooseAdapter.js:630:51)
    at /node_modules/json-api/build/src/db-adapters/Mongoose/MongooseAdapter.js:641:25
    at Schema.eachPath (/node_modules/mongoose/lib/schema.js:506:5)
    at Function.getStandardizedSchema (/node_modules/json-api/build/src/db-adapters/Mongoose/MongooseAdapter.js:636:20)
    at DocumentationController.getTypeInfo (/node_modules/json-api/build/src/controllers/Documentation.js:117:40)
    at /node_modules/json-api/build/src/controllers/Documentation.js:69:39
    at Array.forEach (native)
    at new DocumentationController (/node_modules/json-api/build/src/controllers/Documentation.js:68:27)
    ...
```

When I remove the `message` property it works fine.

**Comment by ethanresnick** [comment]
So, there are two distinct things going on here. 

The first is that the json-api library is trying to read your schema to automatically generate documentation, and, as you suggest, it's having trouble with arrays of embedded objects. That's definitely a bug/missing feature, and I'll keep this issue open as a reminder to add it.

But there's a deeper problem here, which is that the [JSON API](http://jsonapi.org/) specification, which this library merely implements, doesn't (yet) support nesting data in the way you're trying to do. Here's the relevant paragraph from the spec:

> Complex data structures involving JSON objects and arrays are allowed as attribute values. However, any object that constitutes or is contained in an attribute **MUST NOT** contain a `relationships` or `links` member, as those members are reserved by this specification for future use.

In other words, having a `messages` attribute that holds an array of objects is ok, but those objects can't have a relationship to other model(s), so the `author` relationship to `User` is impossible.

There has been [much discussion](https://github.com/json-api/json-api/issues/383) about allowing relationships in object attributes, and it may well happen—but not for at least a few months. These spec things move slowly, unfortunately.

For now, therefore, I'd suggest making `message` into it's own model, and having message contain a relationship to the discussion that owns it (or vice-versa). Then everything will work smoothly with this library and the specification.
